backend/tests/test_recovery_scoring.py

```
@@ -0,0 +1,592 @@
+        a = score_recovery_paths(paths, {"strategy_preference": "income-first"})
+        b = score_recovery_paths(paths, {"strategy_preference": "income-first"})
+        assert json.dumps(a, sort_keys=True) == json.dumps(b, sort_keys=True)
+
+
+# ---------------------------------------------------------------------------
+# Static guard: no LLM / HTTP imports in scoring module
+# ---------------------------------------------------------------------------
+
+
+class TestNoExternalCalls:
+    FORBIDDEN = {"httpx", "requests", "openai", "anthropic", "urllib", "socket"}
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)